experiment/selinux/src/selinux.rs

```diff
@@ -0,0 +1,433 @@
+use crate::xattr::*;
+use nix::unistd::gettid;
+use nix::sys::statfs;
+use nix::errno::Errno;
+use std::path::PathBuf;
+use std::fs::File;
+use std::io::{self, Read};
+use std::os::fd::{AsFd, AsRawFd};
+use std::sync::atomic::{AtomicBool, Ordering};
+
+const XATTR_NAME_SELINUX: &str = "security.selinux";
+const ERR_EMPTY_PATH: &str = "empty path";
+static HAVE_THREAD_SELF: AtomicBool = AtomicBool::new(false);
+static INIT_DONE: AtomicBool = AtomicBool::new(false);
+
+// function compatible with setDisabled in go-selinux repo.
```

Comment:
I don't think we have to keep it compatible with the `go-selinux` repo. Is there any specific reason to keep it?

Reply:
In my view, prioritizing Rust-friendliness over maintaining compatibility with `go-selinux` is more important.

Reply:
> In my view, prioritizing Rust-friendliness over maintaining compatibility with go-selinux is more important.

Yeah, agree. I couldn't write Rust-friendliness because I am not so familiar with Rust yet.
Please let me know if you feel weird in terms of Rust. Thank you so much!

Reply:
> Please let me know if you feel weird in terms of Rust. Thank you so much!

Sure. First of all, we can take using `struct` instead of global variables into consideration. WDYT?

Reply:
Sure, as you commented in another thread, I will use struct.
I will fix it after reading experiment/seccomp and other codes.

If we use `struct` here and call this `struct` more than once, I thought maybe the selinux struct is initialized more than once. But the performance caused by this is tiny, so I will use struct here.